src/common/database.c

```diff
@@ -967,6 +967,29 @@ static int _upgrade_library_schema_step(dt_database_t *db, int version)
 
     sqlite3_exec(db->handle, "COMMIT", NULL, NULL, NULL);
     new_version = 15;
+  }
+  else if(version == 15)
+  {
+    sqlite3_exec(db->handle, "BEGIN TRANSACTION", NULL, NULL, NULL);
+    ////////////////////////////// custom image order
+    TRY_EXEC("ALTER TABLE main.images ADD COLUMN position INTEGER",
+             "[init] can't add `position' column to images table in database\n");
+    TRY_EXEC("CREATE INDEX main.image_position_index ON images (position)",
```

Comment:
Aren't the positions unique? Maybe a UNIQUE INDEX would make sense?

Reply:
Done. It is a unique index now. 

My thoughts: I was torn on that. On one hand: it is the right way to model. On the other: we don't gain much with a unique index and in case of any error while position calculation we lose error handling.